doc/user/layouts/partials/create-source/connector/s3/details.html

```diff
@@ -1,22 +1,76 @@
+  you can only entirely replace or delete objects.
+
+The primary effect of this is that we do not handle *updates* to S3 objects, and we may interleave
+multiple object ingestion to speed it up.
+
+[aws-s3]: https://aws.amazon.com/s3/
 
 #### S3 Limitations
 
-* Currently S3 sources do not support Avro-encoded objects.
+* Currently S3 sources do not support Avro- or Protobuf- encoded objects. Implementation issues:
+  [Avro](https://github.com/MaterializeInc/materialize/issues/5596),
+  [Protobuf](https://github.com/MaterializeInc/materialize/issues/5597)
+* Object ingest order is not guaranteed.
 * **All S3 sources are append-only** -- deleted and updated [S3 Objects][] are silently ignored.
 
 [S3 Objects]: https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingObjects.html
 
-#### Scanning S3 Buckets
+#### Object discovery strategies
+
+Materialize has several techniques to discover the objects to download, with [more
+planned][object-specifiers] for the future. Strategies may be specified multiple times within one
+**CREATE SOURCE** statement, so for example this is a legal invocation: `OBJECTS FROM SCAN BUCKET
+'example', SCAN BUCKET 'other', SQS NOTIFICATIONS 'example-notifications'`.
+
+All strategies follow the same basic pattern, `materialized` will:
+
+* Interact with a collection of AWS APIs.
+* Deduplicate objects such that the same object is never downloaded twice.
+* Filter the list of objects discovered via those APIs against the [**MATCHING** clause
+  patterns](#patterns).
+* Treat each object downloaded as a newline-delimited file for the purposes of record
+  delimitation.
 
-The **OBJECTS FROM SCAN BUCKET** syntax causes S3 sources to issue [ListBucket][] requests to collect the
-names of objects to collect. After collecting the list of keys, their names are checked against
-the **MATCHING** clause, if present, before downloading and extracting objects line by line.
+No guarantees are made about the relative order of records sent from the object discovery process
+and the Materialize SQL engine, as usual with SQL you must impose your own order.
+
+[object-specifies]: https://github.com/MaterializeInc/materialize/issues/5502
+
+##### Scanning S3 Buckets
+
+The **OBJECTS FROM SCAN BUCKET** syntax causes S3 sources to issue [ListBucket][] requests to
+collect the names of objects to collect. After collecting the list of keys, their names are
+checked against the **MATCHING** clause, if present, before downloading and extracting objects
+line by line.
 
 [ListBucket]: https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html
 
+##### Listening to SQS Notifications
+
+AWS S3 has a built-in method for notifying downstream applications of bucket modification, the
+[Event Notification API][notification-aws]. For the purpose of `materialized` the only interesting
+modifications are object creation, aka the `s3:ObjectCreated:*` event namespace. Following [the
```

Comment:
Insert comma after `materialized`

I would make this "Follow the AWS Tutorial to configure a bucket for exactly this namespace."